gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -852,11 +851,64 @@ public void run() {
       catch (Exception e) {
         if (getLoggerI18n().warningEnabled()) {
           getLoggerI18n().warning(LocalizedStrings.DEBUG,
-              "OldEntriesCleanerThread : Error occured while cleaning the oldentries map.Actual " +
+              "OldEntriesCleanerThread : Error occured while cleaning the oldentries map. Actual " +
                   "Exception:", e);
         }
       }
     }
+
+    boolean notRequiredByAnyTx(BlockingQueue<RegionEntry> queue,
+        Region region, RegionEntry re) {
+      int myVersion = re.getVersionStamp().getEntryVersion();
+      Set<TXId> txIds = new HashSet<TXId>();
+
+      for (TXStateProxy txProxy : getTxManager().getHostedTransactionsInProgress()) {
+        TXState txState = txProxy.getLocalTXState();
+        if ((txState != null && !txState.isCommitted() && TXState.checkEntryInSnapshot
+            (txState, region, re))) {
+          txIds.add(txState.getTransactionId());
+        }
+      }
+
+      for (RegionEntry regionEntry : queue) {
+        if (regionEntry == re)
+          continue;
+        Set<TXId> othersTxIds = new HashSet<TXId>();
+        for (TXStateProxy txProxy : getTxManager().getHostedTransactionsInProgress()) {
+          TXState txState = txProxy.getLocalTXState();
+          if ((txState != null && !txState.isCommitted() && TXState.checkEntryInSnapshot
+              (txState, region, regionEntry))) {
+            othersTxIds.add(txState.getTransactionId());
+          }
+        }
+
+        if (txIds.size() == othersTxIds.size() &&
+            txIds.containsAll(othersTxIds) && othersTxIds.containsAll(txIds)
+            && regionEntry.getVersionStamp().getEntryVersion() > myVersion) {
+          return true;
+        }
+      }
+
+      // in the end check with the entry in region
+      RegionEntry entryInRegion = ((LocalRegion)region).entries.getEntry(re.getKey());
+      Set<TXId> othersTxIds = new HashSet<TXId>();
```

Comment:
Region entry could be null or destroyed?

Reply:
Added null check.